lms/djangoapps/teacher_dashboard/views.py

```diff
@@ -34,52 +35,49 @@ def dashboard_view(request, course_id):
-    if not course_id:
-        response = []
-    else:
-        course_key = SlashSeparatedCourseKey.from_deprecated_string(course_id)
-        with modulestore().bulk_operations(course_key):
-            course = get_course_by_id(course_key, depth=2)
-
-        # CCX overrides doesn't work outside of Courseware, so we have to get them via
-        # get_override_for_ccx.
-        # Next 3 lines work well with both: ccx and simple courses.
-        ccx = get_ccx_from_ccx_locator(course_key)
-        passports = get_override_for_ccx(ccx, course, 'lti_passports', course.lti_passports)[:]
-        consumer_keys = [LtiPassport(passport_str).consumer_key for passport_str in passports]
-
-        url = settings.LABSTER_ENDPOINTS.get("licenses")
-        response = _send_request(
-            url,
-            method="POST",
-            data=json.dumps({"consumer_keys": consumer_keys}),
-        )
+    course_key = SlashSeparatedCourseKey.from_deprecated_string(course_id)
+    with modulestore().bulk_operations(course_key):
+        course = get_course_by_id(course_key, depth=None)
+
+    # CCX overrides doesn't work outside of Courseware, so we have to get them via
+    # get_override_for_ccx.
+    # Next 3 lines work well with both: ccx and simple courses.
+    ccx = get_ccx_from_ccx_locator(course_key)
+    passports = get_override_for_ccx(ccx, course, 'lti_passports', course.lti_passports)[:]
+    consumer_keys = [LtiPassport(passport_str).consumer_key for passport_str in passports]
+
+    url = settings.LABSTER_ENDPOINTS.get("licenses")
+    response = _send_request(
+        url,
+        method="POST",
+        data=json.dumps({"consumer_keys": consumer_keys}),
+    )
+
     return HttpResponse(response, content_type="application/json")
 
 
 @login_required
-def simulations_api_call(request, license_pk):
+def simulations_api_call(request, course_id, license_pk):
     url = settings.LABSTER_ENDPOINTS.get('simulations').format(license_pk)
     response = _send_request(url)
     return HttpResponse(response, content_type="application/json")
 
 
 @login_required
-def students_api_call(request, license_pk, simulation_pk):
+def students_api_call(request, course_id, license_pk, simulation_pk):
     url = settings.LABSTER_ENDPOINTS.get('students').format(license_pk, simulation_pk)
     content = _send_request(url)
     return HttpResponse(content, content_type="application/json")
 
 
 @login_required
-def attempts_api_call(request, license_pk, simulation_pk):
+def attempts_api_call(request, course_id, license_pk, simulation_pk):
     url = settings.LABSTER_ENDPOINTS.get('attempts').format(license_pk, simulation_pk)
     content = _send_request(url, headers={"accept": "text/csv"})
     response = HttpResponse(content, content_type="text/csv")
-    response['Content-Disposition'] = 'attachment; filename="export_student_results.csv"'
+    response['Content-Disposition'] = 'attachment; filename="{0}-summary.csv"'.format(course_id)
```

Comment:
can be course_id like "MC/MC/courseware"? if so, will we have a problems with create this file 
MC/MC/courseware-summary.csv ?